web3c/key_manager.js

```diff
@@ -1,7 +1,12 @@
 // Stores known short and longterm keys for contracts,
 // refreshing + validating short-term keys as needed.
 const nacl = require('tweetnacl');
+const hash = require('js-sha512').sha512_256;
```

Comment:
sha512 is supported in [subtle](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/digest) and [node](https://nodejs.org/api/crypto.html#crypto_class_hash) prefer having this exposed in the current crypto folder, rather than introducing the new dependency.

Reply:
This isn't sha 512. This is sha 512/256.

Reply:
Sha 512/256 isn't supported in subtle and node from what I can tell.

Reply:
Sha 512/256 is sha512 truncate to the first 256 bits of the final hash. You'll be doing the same thing as in
https://github.com/oasislabs/web3c.js/blob/master/crypto/subtle/siv_ctr.js#L40

where we compute sha256/128

Reply:
"Sha 512/256 is sha512 truncate to the first 256 bits of the final hash." this is not true.

Reply:
are we not able to set the different IV explicitly in subtle/node?

Reply:
Not that I can tell. Alternatively we can change the hash function at the platform level, since key management is isolated enough to where it shouldn't matter--though I'd rather punt that for a later date.

Reply:
@Yawning - thoughts on if sha512/256 is the right way to go?